File: offer/27/main.go
package main

// Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func mirrorTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	stack := []*TreeNode{root}

	for len(stack) != 0 {
		node := stack[0]
		stack = stack[1:]

		if node.Right != nil {
			stack = append(stack, node.Right)
		}
		if node.Left != nil {
			stack = append(stack, node.Left)
		}
		node.Left, node.Right = node.Right, node.Left
	}
	return root
}

File: offer/27/main.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def mirrorTree(self, root: TreeNode) -> TreeNode:
        def swap(node: TreeNode):
            if not node:
                return

            node.left, node.right = node.right, node.left
            swap(root.left)
            swap(root.right)

        swap(root)
        return root
